samples/opengl/depthMapTriangulation/image_triangulator.cpp

```diff
@@ -0,0 +1,949 @@
+    ImageView<MyMesh::VertexHandle> selected_vertices_imV =
+        ImageView<MyMesh::VertexHandle>(settings.image_height, settings.image_width, selected_vertices_vec.data());
+
+    select_vertices_for_RQT(mesh_out, unprojected_imV, selected_vertices_imV);
+
+
+    // actually triangulate the mesh using all the selected vertices
+    triangulate_RQT_selected_vertices(mesh_out, selected_vertices_imV);
+
+
+    // --- post computation ---
+
+    // activate status so that items can be deleted
+    mesh_out.request_face_status();
+    mesh_out.request_edge_status();
+    mesh_out.request_vertex_status();
+
+    // cleaning up the result (not calling these can kill circulators later on for some reason)
+    mesh_out.delete_isolated_vertices();
+    mesh_out.garbage_collection();
+
+    mesh_out.release_face_status();
+    mesh_out.release_edge_status();
+    mesh_out.release_vertex_status();
+}
+
+
+// ----  PRIVATE  ----
+void Saiga::RQT_Triangulator::add_face_to_mesh(MyMesh& mesh, MyMesh::VertexHandle vh1, MyMesh::VertexHandle vh2,
+                                               MyMesh::VertexHandle vh3)
+{
+    if (mesh.point(vh1)[2] == settings.broken_values || mesh.point(vh2)[2] == settings.broken_values || mesh.point(vh3)[2] == settings.broken_values) return;
+
+    mesh.add_face(vh1, vh2, vh3);
+}
+
+int Saiga::RQT_Triangulator::get_closest_RQT_side_length(int height, int width)
+{
+    int max_image_side_length = std::max(height, width);
+    // the width has to be a power of 2 with 1 added on top
+    if (((max_image_side_length - 1) & (max_image_side_length - 2)) == 0)
+    {
+        // The width is already as needed
+    }
+    else if ((max_image_side_length & (max_image_side_length - 1)) == 0)
+    {
+        // The width is a power of 2, so only 1 needs to be added
+        ++max_image_side_length;
+    }
+    else
+    {
+        // The width is neither a power of 2 nor a power of two plus 1.
+        // --> Find the next power of two and add 1
+        max_image_side_length = log2(max_image_side_length);
+        ++max_image_side_length;
+        max_image_side_length = pow(2, max_image_side_length);
+        ++max_image_side_length;
+    }
+
+    return max_image_side_length;
+}
+
+void Saiga::RQT_Triangulator::create_dependency_graph()
+{
+    // level 0 isn't captured by log, so add it manually
+    int total_levels_without_zero = log2(RQT_side_len - 1);
+    int total_levels              = total_levels_without_zero + 1;
+
+    dependency_graph_vector = std::vector<std::vector<Point2D>>(RQT_side_len * RQT_side_len);
+    dependency_graph = ImageView<std::vector<Point2D>>(RQT_side_len, RQT_side_len, dependency_graph_vector.data());
+
+    // add something to the level 0 vertices so they will be skipped in the upcoming loop over the other levels
+    Point2D tmp_point(0, 0);
+    std::vector<Point2D> tmp_vec = std::vector<Point2D>();
+    tmp_vec.push_back(tmp_point);
```

Comment:
Do you have upper size estimates for these vectors?
So we can call `reserve` on them